proposals/0021-root-signature-in-clang.md

```diff
@@ -0,0 +1,239 @@
+<!-- {% raw %} -->
+
+# Implementation of RootSignature in clang
+
+* Proposal: [0021](0021-root-signature-in-clang.md)
+* Author(s): [Xiang Li](https://github.com/python3kgae)
+* Sponsor: TBD
+* Status: **Under Consideration**
+
+
+## Introduction
+
+In HLSL, the [root signature](https://learn.microsoft.com/en-us/windows/win32/direct3d12/root-signatures)
+defines what types of resources are bound to the graphics pipeline.
+
+A root signature can be specified in HLSL as a [string](https://learn.microsoft.com/en-us/windows/win32/direct3d12/specifying-root-signatures-in-hlsl#an-example-hlsl-root-signature).
+The string contains a collection of comma-separated clauses that describe root
+signature constituent components.
+
+## Motivation
+
+To support root signature in clang-dxc.
+
+## Requirements
+
+1) Clang-dxc should parse existing root signature grammars and provide 
+diagnostics.
+2) Clang-dxc should translate textual root signatures into binary encoded 
+formats.
+3) Clang-dxc should support reading and writing binary encoded root 
+ignatures to and from DXContainer files.
+4) Clang-dxc should validate root signatures before emitting them.
+5) Clang-dxc should have a comprehensive test infrastructure to test all 
```

Comment:
nit: I don't really like calling these all `clang-dxc`. `clang-dxc` is a specific software component, it is the binary (or symlink) for the Clang DXC-compatible driver. We don't actually want to do this in the driver, and some of it shouldn't even necessarily be exposed through the driver.

Reply:
Updated

Reply:
@llvm-beanz - ok to resolve this conversation?